File: Reverse Linked List - Leetcode 206/Reverse Linked List - Leetcode 206.cpp
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* prev = nullptr;
        ListNode* cur = head;

        while (cur != nullptr) {
            ListNode* temp = cur->next;
            cur->next = prev;
            prev = cur;
            cur = temp;
        }

        return prev;
    }
};

File: Reverse Linked List - Leetcode 206/Reverse Linked List - Leetcode 206.py
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        cur = head
        prev = None
        
        while cur:
            temp = cur.next
            cur.next = prev
            prev = cur
            cur = temp
        
        return prev

# Time Complexity: O(n)
# Space Complexity: O(1)
